draft-packages/modal/KaizenDraft/Modal/Presets/InformationModal.tsx

```diff
@@ -34,75 +36,97 @@ type InformationModal = React.FunctionComponent<InformationModalProps>
 const InformationModal = ({
   isOpen,
   title,
-  onDismiss,
   onConfirm,
   confirmLabel = "Confirm",
+  confirmWorking,
   automationId,
   renderBackground,
   children,
   contentHeader,
   image,
   ...props
-}: InformationModalProps) => (
-  <GenericModal
-    isOpen={isOpen}
-    onEscapeKeyup={onDismiss}
-    onOutsideModalClick={onDismiss}
-    automationId={automationId}
-  >
-    <div className={styles.modal}>
-      {renderBackground && renderBackground()}
-      <ModalHeader unpadded onDismiss={onDismiss}>
-        <div className={styles.header}>
-          <ModalAccessibleLabel>
-            <Heading variant="heading-2" tag="h1">
-              <Box pb={0.5}>{title}</Box>
-            </Heading>
-          </ModalAccessibleLabel>
-        </div>
-      </ModalHeader>
-      <Divider variant="content" />
-      {contentHeader && (
-        <Box mt={0.5} mx={1} mb={1}>
-          {contentHeader}
-        </Box>
-      )}
-      <div className={styles.contentLayout}>
-        <div className={styles.content}>
-          {children}
-          {onConfirm != null && (
-            <div
-              className={
-                props.secondaryLabel
-                  ? styles.footerWithSecondaryAction
-                  : styles.footer
-              }
-            >
-              <ModalFooter
-                unpadded
-                alignStart
-                variant={image ? "information" : undefined}
-                actions={[
-                  { label: confirmLabel, onClick: onConfirm },
-                  ...(props.secondaryLabel
-                    ? [
-                        {
-                          label: props.secondaryLabel,
-                          onClick: props.onSecondaryAction,
-                        },
-                      ]
-                    : []),
-                ]}
-                appearance={"primary"}
-                automationId={automationId}
-              />
-            </div>
-          )}
+}: InformationModalProps) => {
+  const onDismiss = confirmWorking ? undefined : props.onDismiss
+
+  const footerActions: ButtonProps[] = []
+
+  const workingProps = confirmWorking
+    ? {
+        working: true,
+        workingLabel: confirmWorking.label,
+        workingLabelHidden: confirmWorking.labelHidden,
+      }
+    : {}
+
+  if (onConfirm) {
+    const confirmAction = { label: confirmLabel, onClick: onConfirm }
+
+    footerActions.push({ ...confirmAction, ...workingProps })
+  }
+
```

Comment:
The `ConfirmationModal` has a separate footerActions included

`  footerActions.push({
    label: dismissLabel,
    onClick: onDismiss,
    disabled: !!confirmWorking,
  })
`

Unsure if this is something that is required on the InformationModal?